### PR TITLE
[DPE-4976] Add workflow for nightly scheduled tests with juju 3.6

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
We want to run integration tests on a nightly basis with juju 3.6-beta

## Solution
Upgrade to dpw v21.0.0 and add juju 3.6 into the matrix for tests. Also, upgrade libjuju to `^3.5.2.0`

### CODE REVIEW

File: .github/workflows/juju_3_6_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - lint
+      - build
+    uses: canonical/data-platform-workflows/.github/workflows/integration_test_charm.yaml@v18.0.0
+    with:
+      artifact-prefix: ${{ needs.build.outputs.artifact-prefix }}
+      architecture: ${{ matrix.architecture }}
+      cloud: microk8s
+      juju-snap-channel: 3.6/beta
+      libjuju-version-constraint: 3.5.2.0
+      microk8s-snap-channel: 1.28-strict/stable
```
Comment (taurus-forever) [inline .github/workflows/juju_3_6_ci.yaml:41]: What about to add renovate label as discussed in MM?

**Review by taurus-forever — APPROVED** [review]
LGTM, one nit.

**Comment by shayancanonical** [comment]
All tests pass on juju 3.6 (https://github.com/canonical/mysql-k8s-operator/actions/runs/10507249164)

File: .github/workflows/juju_3_6_ci.yaml (hunk 1 above)
Comment (shayancanonical) [reply .github/workflows/juju_3_6_ci.yaml]: added renovate label in [5d740e3](https://github.com/canonical/mysql-k8s-operator/pull/490/commits/5d740e3252c1e9de6b2da8e5b40ef7c55c112748)

**Review by paulomach — APPROVED** [review]

**Pull request merged**